subprojects/build-cache/src/main/java/org/gradle/caching/internal/controller/DefaultNextGenBuildCacheAccess.java

```diff
@@ -90,6 +88,7 @@ public <T> void load(Map<BuildCacheKey, T> entries, LoadHandler<T> handler) {
                 }
                 if (!foundLocally && remote.canLoad()) {
                     // TODO Improve error handling
+                    handler.startLoadOperation(key);
```

Comment:
Does that start multiple load operations if multiple entries are requested from remote cache? Similar also for store

Reply:
This is what `OnDemandBuildOperationWrapper` covers: it only starts the build operation at most once, and only closes it if it was started.

Reply:
Maybe `ensureOperationStarted()` is a better name, lemme rename.